libraries/rush-lib/src/logic/operations/OperationMetadataManager.ts

```diff
@@ -112,31 +141,46 @@ export class OperationMetadataManager {
 
   public async tryRestoreAsync({
     terminal,
-    logPath,
-    errorLogPath
+    terminalProvider,
+    errorLogPath,
+    logChunksPath
   }: {
+    terminalProvider: ITerminalProvider;
     terminal: ITerminal;
     logPath: string;
     errorLogPath: string;
+    logChunksPath: string;
   }): Promise<void> {
     await this.stateFile.tryRestoreAsync();
 
-    // Append cached log into current log file
-    terminal.writeLine(`Restoring cached log file at ${this._logPath}`);
     let logReadStream: fs.ReadStream | undefined;
     try {
-      logReadStream = fs.createReadStream(this._logPath, {
-        encoding: 'utf-8'
-      });
-      for await (const data of logReadStream) {
-        terminal.write(data);
+      if (await FileSystem.existsAsync(logChunksPath)) {
+        const logChunks: ILogChunkStorage = (await JsonFile.loadAndValidateAsync(
+          logChunksPath,
+          OperationMetadataManager._logChunkJsonSchema
```

Comment:
If this file is completely internal to Rush, I'm not sure there's much value to validating it every time it's read.

Reply:
Makes sense, I'll remove.